### PR TITLE
Update the implementation of ssim() loss function, reduce the computational complexity from O(n) to O(1) since create_window() is called once once in main function

### PR DESCRIPTION
**Pull request by tryhiseyyysum**
In the loss_utils.py: Updated the function ssim(), the optimized version of ssim() is ssim_optimized(), which reduce the computational complexity. After modified, the function create_window() just need to be called once in the main function in train.py, no need to be called in ssim() in every iteration.

In the train.py: Updated the main function, call the create_window() function for just once, since the window_size is fixed to 11 and the channel is fixed to 3 according to the source code. 
Updated the calculation of loss function, call the ssim_optimized(), instead of the original ssim().

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -131,13 +141,18 @@ def training(dataset, opt, pipe, testing_iterations, saving_iterations, checkpoi
...
+
+
 def prepare_output_and_logger(args):    
     if not args.model_path:
         if os.getenv('OAR_JOB_ID'):
             unique_str=os.getenv('OAR_JOB_ID')
         else:
             unique_str = str(uuid.uuid4())
-        args.model_path = os.path.join("./output/", unique_str[0:10])
+        args.model_path = os.path.join("/mnt/data1/3dgs_modify_output/", unique_str[0:10])
```
Comment (AsherJingkongChen) [inline train.py:155]: Not cleaned.

File: train.py [hunk 2]
```diff
@@ -191,6 +206,11 @@ def training_report(tb_writer, iteration, Ll1, loss, l1_loss, elapsed, testing_i
         torch.cuda.empty_cache()
 
 if __name__ == "__main__":
+    #----------------------create window------------------
+    window_size=11
+    channel=3
+    window=create_window(window_size, channel)
```
Comment (AsherJingkongChen) [inline train.py:212]: Your window is not passed anywhere. It is a local variable in train.py

File: utils/loss_utils.py [hunk 3]
```diff
@@ -62,3 +63,12 @@ def _ssim(img1, img2, window, window_size, channel, size_average=True):
...
 
+#-----------------modify------------------------------------
+def ssim_optimized(img1, img2, window=None, window_size=11, size_average=True):
+    channel = img1.size(-3)
+    if window is None:
+        window = create_window(window_size, channel).to(img1.device).type_as(img1)
+    if img1.is_cuda:
+        window = window.cuda(img1.get_device())
+    window = window.type_as(img1)
+    return _ssim(img1, img2, window, window_size, channel, size_average)
```
Comment (AsherJingkongChen) [inline utils/loss_utils.py:74]: Your window is always none.

**Review by AsherJingkongChen — CHANGES_REQUESTED** [review]
